gcp/src/main/java/org/apache/iceberg/gcp/biglake/BigLakeCatalog.java

```diff
@@ -0,0 +1,398 @@
+    String fromDbId = getDatabaseId(from.namespace());
+    String toDbId = getDatabaseId(to.namespace());
+
+    Preconditions.checkArgument(
+        fromDbId.equals(toDbId), "New table name must be in the same database");
+    client.renameTable(getTableName(fromDbId, from.name()), getTableName(toDbId, to.name()));
+  }
+
+  @Override
+  public void createNamespace(Namespace namespace, Map<String, String> metadata) {
+    if (namespace.levels().length == 0) {
+      // Used by `CREATE NAMESPACE <catalog>`. Create a BLMS catalog linked with Iceberg catalog.
+      client.createCatalog(catalogName, Catalog.getDefaultInstance());
+      LOG.info("Created BigLake catalog: {}", catalogName.toString());
+    } else if (namespace.levels().length == 1) {
+      // Create a database.
+      String dbId = namespace.level(0);
+      Database.Builder builder = Database.newBuilder().setType(Database.Type.HIVE);
+      builder
+          .getHiveOptionsBuilder()
+          .putAllParameters(metadata)
+          .setLocationUri(getDatabaseLocation(dbId));
+
+      client.createDatabase(DatabaseName.of(projectId, location, catalogId, dbId), builder.build());
+    } else {
+      throw new IllegalArgumentException(invalidNamespaceMessage(namespace));
+    }
+  }
+
+  @Override
+  public List<Namespace> listNamespaces(Namespace namespace) {
+    if (namespace.levels().length != 0) {
+      // BLMS does not support namespaces under database or tables, returns empty.
+      // It is called when dropping a namespace to make sure it's empty (listTables is called as
+      // well), returns empty to unblock deletion.
+      return ImmutableList.of();
+    }
+
+    return Streams.stream(client.listDatabases(catalogName))
+        .map(BigLakeCatalog::getNamespace)
+        .collect(ImmutableList.toImmutableList());
+  }
+
+  @Override
+  public boolean dropNamespace(Namespace namespace) {
+    try {
+      if (namespace.levels().length == 0) {
+        // Used by `DROP NAMESPACE <catalog>`. Deletes the BLMS catalog linked by Iceberg catalog.
+        client.deleteCatalog(catalogName);
+        LOG.info("Deleted BigLake catalog: {}", catalogName.toString());
+      } else if (namespace.levels().length == 1) {
+        client.deleteDatabase(getDatabaseName(namespace));
+        // We don't delete the data file folder for safety. It aligns with HMS's default behavior.
+        // We can support database or catalog level config controlling file deletion in future.
+      } else {
+        throw new IllegalArgumentException(invalidNamespaceMessage(namespace));
+      }
+    } catch (NoSuchNamespaceException e) {
+      LOG.warn("Dropping namespace failed", e);
+      return false;
+    }
+
+    return true;
+  }
+
+  @Override
+  public boolean setProperties(Namespace namespace, Map<String, String> properties) {
+    Database.Builder builder;
+    try {
+      builder = getDatabase(namespace).toBuilder();
+    } catch (IllegalArgumentException e) {
```

Comment:
I don't think this is needed. If the database can't be loaded, then this should throw `NoSuchNamespaceException`.

Reply:
It is because invalid namespace throws IllegalArgumentException. I have changed it to throw NoSuchNamespaceException and removed this catch.